receiver/googlecloudspannerreceiver/internal/filterfactory/itemfilterfactory_test.go

```diff
@@ -0,0 +1,153 @@
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package filterfactory
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"go.uber.org/zap/zaptest"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/googlecloudspannerreceiver/internal/filter"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/googlecloudspannerreceiver/internal/metadata"
+)
+
+func TestNewItemFilterFactory(t *testing.T) {
+	logger := zaptest.NewLogger(t)
+	metricPrefixes := []string{prefix1, prefix2}
+	prefixHighCardinality := []bool{true, true}
+	metadataItems := generateMetadataItems(metricPrefixes, prefixHighCardinality)
+	testCases := map[string]struct {
+		totalLimit  int
+		expectError bool
+	}{
+		"Total limit is zero":                          {0, false},
+		"Total limit is positive":                      {200 * defaultMetricDataPointsAmountInPeriod, false},
+		"Total limit is lover then product of amounts": {3, true},
+		"Error when limit by timestamp is lower than 1 for high cardinality groups": {20, true},
+	}
+
+	for name, testCase := range testCases {
+		t.Run(name, func(t *testing.T) {
+			config := &ItemFilterFactoryConfig{
+				MetadataItems:  metadataItems,
+				TotalLimit:     testCase.totalLimit,
+				ProjectAmount:  1,
+				InstanceAmount: 2,
+				DatabaseAmount: 5,
+			}
+
+			factory, err := NewItemFilterFactory(logger, config)
+
+			if testCase.expectError {
+				require.Error(t, err)
+				require.Nil(t, factory)
+			} else {
+				require.NoError(t, err)
+			}
+		})
+	}
+}
+
+func TestItemFilterFactoryConfig_Validate(t *testing.T) {
+	testCases := map[string]struct {
+		metadataItems  []*metadata.MetricsMetadata
+		totalLimit     int
+		projectAmount  int
+		instanceAmount int
+		databaseAmount int
+		expectError    bool
+	}{
+		"No metadata items":                            {[]*metadata.MetricsMetadata{}, 10, 1, 1, 1, true},
+		"Total limit is zero":                          {[]*metadata.MetricsMetadata{{}}, 0, 1, 1, 1, false},
+		"Total limit is lover then product of amounts": {[]*metadata.MetricsMetadata{{}}, 3, 1, 2, 3, true},
+	}
+
+	for name, testCase := range testCases {
+		t.Run(name, func(t *testing.T) {
+			config := &ItemFilterFactoryConfig{
+				MetadataItems:  testCase.metadataItems,
+				TotalLimit:     testCase.totalLimit,
+				ProjectAmount:  testCase.projectAmount,
+				InstanceAmount: testCase.instanceAmount,
+				DatabaseAmount: testCase.databaseAmount,
+			}
+
+			err := config.validate()
+
+			if testCase.expectError {
+				require.Error(t, err)
+			} else {
+				require.NoError(t, err)
+			}
```

Comment:
```suggestion
assert.Equal(t, testCase.ExpectError, err != nil)
```

Reply:
Not sure if I have the bool check the write way, but I am hoping it conveys the idea that these can be simplified

Reply:
I prefer having `if` statement instead of this assertion. In case of `if` approach it is more clear for me and others to understand that I am (not)expecting error here (stated explicitly).